omod/src/test/java/org/openmrs/module/fhir2/narrative/DefaultNarrativeTemplateTest.java

```diff
@@ -0,0 +1,230 @@
+		route.addCoding().setDisplay("Oral Route").setSystem("http://snomed.info/sct").setCode("26643006");
+		dosage.setRoute(route);
+		CodeableConcept method = new CodeableConcept();
+		method.addCoding().setDisplay("Swallow - dosing instruction imperative (qualifier value)")
+		        .setSystem("http://snomed.info/sct").setCode("421521009");
+		dosage.setMethod(method);
+		
+		CodeableConcept doseandrateType = new CodeableConcept();
+		doseandrateType.addCoding().setCode("ordered").setDisplay("Ordered")
+		        .setSystem("http://terminology.hl7.org/CodeSystem/dose-rate-type");
+		Range doseRange = new Range();
+		doseRange.setLow(new Quantity().setValue(1).setUnit("TAB")
+		        .setSystem("http://terminology.hl7.org/CodeSystem/v3-orderableDrugForm").setCode("TAB"));
+		doseRange.setHigh(new Quantity().setValue(2).setUnit("TAB")
+		        .setSystem("http://terminology.hl7.org/CodeSystem/v3-orderableDrugForm").setCode("TAB"));
+		dosage.addDoseAndRate().setDose(doseRange).setType(doseandrateType);
+		
+		medicationRequest.addDosageInstruction(dosage);
+		
+		Duration duration = new Duration();
+		duration.setValue(10).setUnit("days").setSystem("http://unitsofmeasure.org").setCode("d");
+		medicationRequest.setDispenseRequest(new MedicationRequest.MedicationRequestDispenseRequestComponent()
+		        .setValidityPeriod(new Period().setStart(new Date()).setEnd(new Date())).setNumberOfRepeatsAllowed(0)
+		        .setQuantity(new Quantity().setValue(30).setUnit("TAB")
+		                .setSystem("http://terminology.hl7.org/CodeSystem/v3-orderableDrugForm").setCode("TAB"))
+		        .setExpectedSupplyDuration(duration).setPerformer(new Reference().setReference("Practitioner/f001"))
+		        .setInitialFill(new MedicationRequest.MedicationRequestDispenseRequestInitialFillComponent()
+		                .setQuantity(new Quantity().setValue(123).setSystem("something")).setDuration(duration)));
+		
+		CodeableConcept substitutionReason = new CodeableConcept();
+		substitutionReason.addCoding().setSystem("http://terminology.hl7.org/CodeSystem/v3-ActReason").setCode("FP")
+		        .setDisplay("formulary policy");
+		medicationRequest.setSubstitution(new MedicationRequest.MedicationRequestSubstitutionComponent()
+		        .setAllowed(new BooleanType().setValue(true)).setReason(substitutionReason));
+		
+		medicationRequest.addDetectedIssue().setReference("DetectedIssue/allergy");
+		medicationRequest.addEventHistory().setReference("#signature").setDisplay("Author's Signature");
+		
+		assertThat(testHelper(new MedicationRequest()).getDivAsString(), notNullValue());
+		assertThat(testHelper(medicationRequest).getDivAsString(), notNullValue());
+	}
+	
+	public Narrative testHelper(IBaseResource iBaseResource) {
+		Narrative narrative = DefaultNarrativeGeneratorForTest.generateDefaultNarrative(iBaseResource);
+		System.out.println(iBaseResource.getClass().getSimpleName() + " Narrative Status:" + narrative.getStatusAsString());
```

Comment:
Better to use logger if you really need or you can remove the `System.out.println` from the code

Reply:
Thanks for reviewing @suthagar23 :) 
Actually I've put this just for convenience in checking the output for now. It isn't intended to be a part of the final commit.

Reply:
Rather than using a logger (which is a good idea if we were going to output anything) or... manually reading the output, use assertions to check that these are the expected values. That's why they exist.